owncloudApp/src/main/java/com/owncloud/android/usecases/transfers/uploads/UploadFileFromContentUriUseCase.kt

```diff
@@ -52,13 +58,22 @@ class UploadFileFromContentUriUseCase(
             .build()
 
         val uploadFileFromContentUriWorker = OneTimeWorkRequestBuilder<UploadFileFromContentUriWorker>()
-            .setInputData(inputData)
+            .setInputData(inputDataUploadFileFromContentUriWorker)
             .setConstraints(constraints)
             .addTag(params.accountName)
             .addTag(params.uploadIdInStorageManager.toString())
             .build()
 
-        workManager.enqueue(uploadFileFromContentUriWorker)
+        var workContinuation = workManager.beginWith(uploadFileFromContentUriWorker)
+        val behavior = UploadBehavior.fromString(params.behavior)
+        if (behavior == UploadBehavior.MOVE) {
+            val removeLocalFileWorker = OneTimeWorkRequestBuilder<RemoveLocalFileWorker>()
+                .setInputData(inputDataRemoveLocalFileWorker)
+                .build()
+            workContinuation = workContinuation.then(removeLocalFileWorker) // File is already uploaded, so the original one can be removed if the behaviour is MOVE
+        }
+        workContinuation.enqueue()
```

Comment:
Is it really that deep? I mean, I think this gets too complex, I'm not even sure if making a `workContinuation` with only 1 work and enqueueing it will work. Wouldn't it be easier if you write inside the `if` block the enqueue of both works and in the `else` block what you had before?

Reply:
Ok, the reason I did it is because I didn't want to repeat code and I thought was better option to make a `workContinuation`. That was working. Also we didn't need it to add an `else`. In any case, let's go with your approach. Could be easy to understand and less complex as you say.